flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/hybrid/HybridSourceSplit.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.base.source.hybrid;
+
+import org.apache.flink.api.connector.source.SourceSplit;
+
+import java.util.Objects;
+
+/** Source split that wraps the actual split type. */
+public class HybridSourceSplit<SplitT extends SourceSplit> implements SourceSplit {
+
+    private final SplitT realSplit;
+    private final int sourceIndex;
+
+    public HybridSourceSplit(int sourceIndex, SplitT realSplit) {
+        this.sourceIndex = sourceIndex;
+        this.realSplit = realSplit;
+    }
+
+    public int sourceIndex() {
+        return this.sourceIndex;
+    }
+
+    public SplitT getWrappedSplit() {
+        return realSplit;
+    }
+
+    @Override
+    public String splitId() {
+        return realSplit.splitId();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        HybridSourceSplit<?> that = (HybridSourceSplit<?>) o;
+        return sourceIndex == that.sourceIndex && realSplit.equals(that.realSplit);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(realSplit, sourceIndex);
+    }
+
+    @Override
+    public String toString() {
+        return "HybridSourceSplit{"
```

Comment:
nit: use `MoreObjects.toStringHelper` to make the formatting a little more reader friendly?